### PR TITLE
CEDS-2400 is EIDR

### PR DESCRIPTION
**Pull request by MaciejRewera**

### CODE REVIEW

File: app/views/components/gds/Styles.scala [hunk 1]
```diff
@@ -19,6 +19,7 @@ package views.components.gds
 object Styles {
 
   val gdsPageLegend = "govuk-fieldset__legend--l"
+  val gdsPageLegendLarge = "govuk-fieldset__legend--xl"
```
Comment (PauloPortugal) [inline app/views/components/gds/Styles.scala]: :+1:

File: test/views/declaration/DeclarantExporterViewSpec.scala [hunk 2]
```diff
@@ -138,7 +137,7 @@ class DeclarantExporterViewSpec extends UnitViewSpec with ExportsTestData with C
         val form = DeclarantIsExporter.form().fill(DeclarantIsExporter(YesNoAnswers.yes))
         val view = createView(form)
 
-        view.getElementById("answer_yes").attr("value") mustBe YesNoAnswers.yes
+        view.getElementById("answer_yes") must beSelected
```
Comment (PauloPortugal) [inline test/views/declaration/DeclarantExporterViewSpec.scala:140]: :+1:

File: app/controllers/navigation/Navigator.scala [hunk 3]
```diff
@@ -267,22 +262,80 @@ object Navigator {
...
   }
 
-  private def exporterDetailsPreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call =
+  private def exporterDetailsClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call =
     if (cacheModel.isEntryIntoDeclarantsRecords)
       controllers.declaration.routes.PersonPresentingGoodsDetailsController.displayPage(mode)
     else
       controllers.declaration.routes.DeclarantExporterController.displayPage(mode)
 
+  private def carrierDetailsPreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call =
```
Comment (PauloPortugal) [inline app/controllers/navigation/Navigator.scala:302]: :+1:

File: app/controllers/navigation/Navigator.scala [hunk 4]
```diff
@@ -309,12 +309,10 @@ object Navigator {
     if (!cacheModel.parties.declarantIsExporter.exists(_.isExporter))
       controllers.declaration.routes.RepresentativeStatusController.displayPage(mode)
     else {
-      if (cacheModel.isExs && cacheModel.parties.consignorDetails.flatMap(_.details.eori.map(_.value)).getOrElse("").nonEmpty)
-        controllers.declaration.routes.ConsignorEoriNumberController.displayPage(mode)
-      else if (cacheModel.isExs && cacheModel.parties.consignorDetails.flatMap(_.details.address).isDefined)
+      if (cacheModel.parties.consignorDetails.flatMap(_.details.address).isDefined)
```
Comment (PauloPortugal) [inline app/controllers/navigation/Navigator.scala:312]: Much better!!! Well spotted!

File: app/controllers/navigation/Navigator.scala [hunk 5]
```diff
@@ -329,10 +327,10 @@ object Navigator {
 
   private def representativeAgentClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode): Call =
     if (cacheModel.isExs) {
-      if (cacheModel.parties.consignorDetails.flatMap(_.details.eori.map(_.value)).getOrElse("").nonEmpty)
-        controllers.declaration.routes.ConsignorEoriNumberController.displayPage(mode)
-      else
+      if (cacheModel.isExs && cacheModel.parties.consignorDetails.flatMap(_.details.address).isDefined)
```
Comment (PauloPortugal) [inline app/controllers/navigation/Navigator.scala:330]: Isn't this check redundant?
Why could you just have 
```
if (cacheModel.parties.consignorDetails.flatMap(_.details.address).isDefined)
```

You did a similar thing on line 312

Comment (PatrykRudnicki) [reply app/controllers/navigation/Navigator.scala:330]: You check `cacheModel.isExs` in line 329 so in line 330 the first part of the check is unnecessary

File: app/forms/common/YesNoAnswer.scala [hunk 6]
```diff
@@ -36,12 +36,11 @@ object YesNoAnswer {
...
-      .verifying("error.yesNo.required", isContainedIn(allowedValues))
-  )(YesNoAnswer.apply)(YesNoAnswer.unapply)
+  private def mapping(fieldName: String, errorKey: String): Mapping[YesNoAnswer] =
+    Forms.mapping(fieldName -> requiredRadio(errorKey).verifying(errorKey, isContainedIn(allowedValues)))(YesNoAnswer.apply)(YesNoAnswer.unapply)
 
   val formId = "YesNo"
 
-  def form(): Form[YesNoAnswer] = Form(mapping)
+  def form(fieldName: String = "yesNo", errorKey: String = "error.yesNo.required"): Form[YesNoAnswer] =
+    Form(mapping(fieldName, errorKey))
```
Comment (PatrykRudnicki) [inline app/forms/common/YesNoAnswer.scala:45]: 👏

File: app/models/declaration/Parties.scala [hunk 7]
```diff
@@ -16,8 +16,9 @@
 
 package models.declaration
 
+import forms.common.YesNoAnswer
 import forms.declaration.consignor.ConsignorDetails
-import forms.declaration.{CarrierDetails, ConsigneeDetails, DeclarantDetails, DeclarantIsExporter, ExporterDetails, IsExs}
+import forms.declaration.{CarrierDetails, ConsigneeDetails, DeclarantDetails, DeclarantIsExporter, ExporterDetails, PersonPresentingGoodsDetails, _}
```
Comment (PatrykRudnicki) [inline app/models/declaration/Parties.scala:21]: Is it not better to just import it with `_`. You have this at the end, so probably rest of the import's parts are unnecessary

File: app/controllers/navigation/Navigator.scala (hunk 5 above)
Comment (MaciejRewera) [reply app/controllers/navigation/Navigator.scala:330]: 👍

**Pull request merged**